src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -1576,18 +1592,11 @@ private static unsafe bool TryNumberToDecimal(ref NumberBuffer number, ref decim
 
             if (c >= '5')
             {
-                // If the next digit is 5, round up if the number is odd or any following digit is non-zero
-                if (c == '5' && (low64 & 1) == 0)
+                if ((c == '5') && ((low64 & 1) == 0) && !number.HasNonZeroTail)
```

Comment:
CC. @pentp on https://github.com/dotnet/coreclr/pull/20707/commits/f9f0034fa516c085bb538b7441c4756e5e86d6ee, since it touches `System.Decimal`.